bot/cogs/source.py

```diff
@@ -10,21 +10,22 @@
 
 
 class SourceConverter(Converter):
-    """Convert an argument into a help command, command, or cog."""
-
-    async def convert(self, ctx: Context, argument: str) -> Union[HelpCommand, Command, Cog]:
-        """
-        Convert argument into source object.
-
-        Order how arguments is checked:
-        1. When argument is `help`, return bot help command
-        2. When argument is valid command, return this command
-        3. When argument is valid Cog, return this Cog
-        4. Otherwise raise `BadArgument` error
-        """
+    """Convert an argument into a help command, tag, command, or cog."""
+
+    async def convert(self, ctx: Context, argument: str) -> Union[HelpCommand, Command, Cog, str]:
+        """Convert argument into source object."""
         if argument.lower() == "help":
             return ctx.bot.help_command
 
+        tags_cog = ctx.bot.get_cog("Tags")
+
+        if argument.lower() in tags_cog._cache:
+            tag = argument.lower()
+            if tags_cog._cache[tag]["restricted_to"] != "developers":
+                return f"bot/resources/tags/{tags_cog._cache[tag]['restricted_to']}/{tag}.md"
+            else:
+                return f"bot/resources/tags/{tag}.md"
```

Comment:
Instead of this, you could edit the tag cog so it stores the file path in its cache.